packages/esm-patient-medications-app/src/add-drug-order/drug-order-form.component.tsx

```diff
@@ -56,91 +57,124 @@ export interface DrugOrderFormProps {
-  unit: z.object(
-    { ...comboSchema },
-    {
-      invalid_type_error: translateFrom(moduleName, 'selectUnitErrorMessage', 'Please select a unit'),
-    },
-  ),
+  const baseSchemaFields = {
+    // t('freeDosageErrorMessage', 'Add free dosage note')
+    freeTextDosage: z.string().refine((value) => !!value, {
+      message: translateFrom(moduleName, 'freeDosageErrorMessage', 'Add free dosage note'),
+    }),
+    // t('dosageRequiredErrorMessage', 'A dosage is required')
+    dosage: z.number({
+      invalid_type_error: translateFrom(moduleName, 'dosageRequiredErrorMessage', 'A dosage is required'),
+    }),
+    // t('selectUnitErrorMessage', 'Please select a unit')
+    unit: z.object(
+      { ...comboSchema },
+      {
+        invalid_type_error: translateFrom(moduleName, 'selectUnitErrorMessage', 'Please select a unit'),
+      },
+    ),
+    // t('selectRouteErrorMessage', 'Please select a route')
+    route: z.object(
+      { ...comboSchema },
+      {
+        invalid_type_error: translateFrom(moduleName, 'selectRouteErrorMessage', 'Please select a route'),
+      },
+    ),
+    patientInstructions: z.string().nullable(),
+    asNeeded: z.boolean(),
+    asNeededCondition: z.string().nullable(),
+    duration: z.number().nullable(),
+    durationUnit: z.object({ ...comboSchema }).nullable(),
+    // t('indicationErrorMessage', 'Please add an indication')
+    indication: z.string().refine((value) => value !== '', {
+      message: translateFrom(moduleName, 'indicationErrorMessage', 'Please add an indication'),
+    }),
+    startDate: z.date(),
+    // t('selectFrequencyErrorMessage', 'Please select a frequency')
+    frequency: z.object(
+      { ...comboSchema },
+      {
+        invalid_type_error: translateFrom(moduleName, 'selectFrequencyErrorMessage', 'Please select a frequency'),
+      },
+    ),
+  };
 
-  // t( 'selectRouteErrorMessage', 'Please select a route' )
-  route: z.object(
-    { ...comboSchema },
-    {
-      invalid_type_error: translateFrom(moduleName, 'selectRouteErrorMessage', 'Please select a route'),
-    },
-  ),
-
-  patientInstructions: z.string().nullable(),
-  asNeeded: z.boolean(),
-  asNeededCondition: z.string().nullable(),
-  duration: z.number().nullable(),
-  durationUnit: z.object({ ...comboSchema }).nullable(),
-  // t( 'pillDispensedErrorMessage', 'The quantity to dispense is required' )
-  pillsDispensed: z.number({
-    invalid_type_error: translateFrom(moduleName, 'pillDispensedErrorMessage', 'The quantity to dispense is required'),
-  }),
-  // t( 'selectQuantityUnitsErrorMessage', 'Dispensing requires a quantity unit' )
-  quantityUnits: z.object(
-    { ...comboSchema },
-    {
-      invalid_type_error: translateFrom(
-        moduleName,
-        'selectQuantityUnitsErrorMessage',
-        'Dispensing requires a quantity unit',
+  const outpatientDrugOrderFields = {
+    // t('pillDispensedErrorMessage', 'The quantity to dispense is required')
+    pillsDispensed: z
```

Comment:
Pills dispensed should be non-negative (and if `requireOutpatientQuantity` is `true` >= 1).